openstudiocore/src/utilities/core/test/Path_GTest.cpp

```diff
@@ -481,3 +481,49 @@ TEST_F(CoreFixture, LastLevelDirectoryWithDot) {
   EXPECT_EQ("A measure with 90.1 dots", toString(lastLevelDir));
   EXPECT_EQ(measure_directory, measure_directory.parent_path() / lastLevelDir);
 }
+
+
+TEST_F(CoreFixture, Path_WithSpecialChars) {
+  // Use a wide string to make sure it's ok when creating and for ulterior comparison with the narrow one
+  std::wstring ws(L"Afolderwithsp�cialCHar#%�/test.osm");
+  openstudio::path pwide;
+  EXPECT_NO_THROW(pwide = path(ws));
+
+  // Make the folder, and touch the test file
+  openstudio::path outfolder = resourcesPath() / toPath("..") / toPath("Testing");
+  openstudio::path pwide_full = outfolder / pwide;
+  makeParentFolder(pwide_full, path(), true);
+  boost::filesystem::ofstream outfile(pwide_full);
+  outfile.close();
+  // Ensure that worked
+  EXPECT_TRUE(openstudio::filesystem::exists(pwide_full));
+
+
+  // Alright, now we try with a regular string with special chars
+  std::string s("Afolderwithsp�cialCHar#%�/test.osm");
```

Comment:
Yeah ruby is definitely produces a different encoding on Windows at least ("Windows-1252"), on Unix it has changed recently to UTF-8 instead of US-ASCII. Cf output results from my docker runs in https://github.com/NREL/OpenStudio-resources/pull/70/files